### DISCUSSION TITLE
hierarchical codelists

### DISCUSSION THREAD

**Discussion by karel-debrabandere-bosa-fgov-be** [OP-TED/eForms-SDK#185] [Q&A] [answered]
Hi,

the docs state the following:
`The few codelists that have a structure, like NUTS and CPV, are indicated as "hierarchical".`
and 
`...most of these are flat lists of terms and their definitions; some, such as the NUTS (Nomenclature of Territorial Units for Statistics) are hierarchical taxonomies.`

where can I find the data that shows this structure? I cannot seem to find any attribute on the cpv codelist items that implies a hierarchical structure.

**Comment by YvesJo** [comment, marked as answer]
Hi,
The genericode files (.gc) provided in the SDK are actually "flat" files, when looking for "hierarchical" information, you may have a look at the other formats available for download on EU Vocabularies: [CPV](https://op.europa.eu/en/web/eu-vocabularies/dataset/-/resource?uri=http://publications.europa.eu/resource/dataset/cpv) and [nuts](https://op.europa.eu/en/web/eu-vocabularies/dataset/-/resource?uri=http://publications.europa.eu/resource/dataset/nuts)
Be aware that only NUTS level 3 codes are expected in eforms.

**Reply by karel-debrabandere-bosa-fgov-be** [reply]
Should this "hierarchical" information not be included in the SDK, since the field codeList value type indeed names it as such?

I am reluctant to introduce external datasources to combine with the codelists provided by the sdk, as these are supposed to be used as primary source of truth.

**Reply by YvesJo** [reply]
This is indeed something nice to have and could be considered in the future. Artefacts from the SDK are sufficient to satisfy the requirements.
